src/content/tools/hooks.md

```diff
@@ -0,0 +1,294 @@
+  native_toolchain_c: any
+
+dev_dependencies:
+  ...
+  ffigen: ^18.0.0
+ ```
+
+### Create a build hook to generate native assets {: #create-hook }
+
+If you want to use a build hook to transparently compile
+native assets (such as C or Rust libraries), which are then
+made available to be called from the Dart code of a package,
+create a `build.dart` script similar to the following:
+
+1.  In your Dart project, create or open `hooks/build.dart`. 
+
+1.  In the `main` method, add a `build` function from
+    `package:hooks/hooks.dart` and use the appropriate
+    toolchain to compile the native library. For example:
+
+    ```dart title="hooks/build.dart" highlightLines=6
+    import 'package:hooks/hooks.dart';
+    import 'package:logging/logging.dart';
+    import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+    void main(List<String> args) async {
+      await build(args, (input, output) async {
+        final packageName = input.packageName;
+        final cbuilder = CBuilder.library(
+          name: packageName,
+          assetName: '$packageName.dart',
+          sources: ['src/$packageName.c'],
+        );
+        await cbuilder.run(
+          input: input,
+          output: output,
+          logger: Logger('')
+            ..level = Level.ALL
+            ..onRecord.listen((record) => print(record.message)),
+        );
+      });
+    }
+    ```
+
+    The `build` function takes two arguments:
+
+    * `input`: (readonly) The input for the hook.
+      Includes information for the hook
+      to produce the right asset type (for example,
+      target OS, target architecture, output directory, and
+      more). For details, see the [`BuildInput`][] class.
+
+    * `output`: (writeonly) The output for the hook. After the
+      build hook reads the input, it produces an asset and
+      then provides what it produced as the output.
+      For details, see the [`BuildOutputBuilder`][] class.
+
+[`BuildInput`]: {{site.pub-api}}/hooks/latest/hooks/BuildInput-class.html
+[`BuildOutputBuilder`]: {{site.pub-api}}/hooks/latest/hooks/BuildOutputBuilder-class.html
+
+### Automatically add assets {: #generate-assets }
+
+Assets are added indirectly when you use the `run`,
+`build`, or `test` command to invoke hooks on your package.
+The resulting assets are stored in the directory that you
+specified in your hooks script. 
+
+### Use assets {: #reference-assets }
+
+Assets are the files that hooks create. Once an asset is
+created, you can reference it in your code and at runtime
+with its asset ID ([`assetId`][]). Asset IDs are structured
+as `package:<package-name>/<asset-name>`. Build hooks may
+only output assets in their own package. `CBuilder` in
+the build hook in the previous example outputs the asset ID
+`package:native_add_library/native_add_library.dart`, and is
+based on the `packageName` and `assetName`.
+
+The following example illustrates how to call the native
+C function `add` from `native_add_library.c`:
```

Comment:
It seems the following code snippet is not necessarily calling the function, just initially binding to it. Perhaps change this to "how to bind" or add an example of calling it as well.